sdk/storage/azblob/blob/client.go

```diff
@@ -242,6 +242,30 @@ func (b *Client) GetTags(ctx context.Context, options *GetTagsOptions) (GetTagsR
 
 }
 
+// SetImmutabilityPolicy operation enables users to set the immutability policy on a blob. Mode defaults to "Unlocked".
+// https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-storage-overview
+func (b *Client) SetImmutabilityPolicy(ctx context.Context, options SetImmutabilityPolicyOptions) (SetImmutabilityPolicyResponse, error) {
```

Comment:
`*SetImmutabilityPolicyOptions`

Reply:
Comment applies to the other APIs.

Reply:
Well wait, that means that everything is optional.  What would that mean to call this API with a `nil` options then?

Reply:
I did not set it as optional since an ExpiryTime must be passed at minimum. Other fields can be nil in the SetImmutabilityPolicyOptions bag

Reply:
I see.  The convention is that all mandatory parameters must be individually specified.  All optional params are in the `Options` type which is passed by pointer.

Reply:
changed!